src/Cache/FileHandler.php

```diff
@@ -60,13 +60,36 @@ public function write(CacheInterface $cache)
     {
         $content = $cache->toJson();
 
+        if (file_exists($this->file)) {
+            if (is_dir($this->file)) {
```

Comment:
so. now the sad part.

I think it could be somehow groomed and simplified.
I could even try to do it.
Yet, after my changes, I have no clue would it work or not.
Why we need to manually check `is_dir` to raise error instead of relying on error after `file_put_contents` ? Maybe we don't?
we shall cover the logic here with tests.

Reply:
Maybe [vfsStream](https://github.com/mikey179/vfsStream) could help writing such tests?

Reply:
Than lets bin this PR until someone has the time to write a fix and tests for this.